cmd/infra/aws/cloudformation/cluster.yaml

```diff
@@ -0,0 +1,488 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: Template for HyperShift Cluster Security Elements (Security Groups & IAM)
+
+Parameters:
+  InfrastructureName:
+    AllowedPattern: ^([a-zA-Z][a-zA-Z0-9\-]{0,26})$
+    MaxLength: 27
+    MinLength: 1
+    ConstraintDescription: Infrastructure name must be alphanumeric, start with a letter, and have a maximum of 27 characters.
+    Description: A short, unique cluster ID used to tag cloud resources and identify items owned or used by the cluster.
+    Type: String
+  BaseDomainHostedZoneId:
+    Description: The Route53 public zone ID which is the parent public zone for the cluster, such as Z21IXYZABCZ2A4.
+    Type: String
+  Subdomain:
+    Description: The Route53 delegated zone for the cluster, such as hypershift.example.com. Omit the trailing period.
+    Type: String
+    Default: "hypershift.example.com"
+
+Metadata:
+  AWS::CloudFormation::Interface:
+    ParameterGroups:
+    - Label:
+        default: "Cluster Information"
+      Parameters:
+      - InfrastructureName
+    ParameterLabels:
+      InfrastructureName:
+        default: "Infrastructure Name"
+
+Mappings:
+  RegionMap:
+    us-east-1:
+      DHCPDomainName: "ec2.internal"
+    us-west-1:
+      DHCPDomainName: "us-west-1.compute.internal"
```

Comment:
Does this mean that we only support `us-east-1` and `us-west-1` as regions right now?

Reply:
Yeah, need to add appropriate mappings for any regions we want to support

Reply:
Can we add all of the public ones? It's a finite list.

Reply:
I went ahead and added the US ones, there might be a way to eliminate this mapping entirely so I added a TODO